extra_data/file_access.py

```diff
@@ -467,25 +467,46 @@ def add_key(key, value):
         self._keys_cache[source] = res
         return res
 
-    def get_one_key(self, source):
+    def get_one_key(self, source, index_group=None):
         """Similar to get_keys(), except it returns only a single key for performance"""
+
+        # Use empty prefix if no index group filter is active.
+        prefix = index_group + '.' if index_group is not None else ''
+
         if source in self._keys_cache:
-            return next(iter(self._keys_cache[source]))
-        elif self._known_keys[source]:
-            return next(iter(self._known_keys[source]))
+            for key in self._keys_cache[source]:
+                if key.startswith(prefix):
+                    return key
```

Comment:
`_keys_cache` should be a complete set, so if we get past this loop without finding any match then we can give up already. `_known_keys` below can be partial, by contrast.

Reply:
Ah, good point. This only allows us to throw an early exception though in the case of a non-existing index group. I added that after the loop.